src/command.ts

```diff
@@ -48,41 +76,111 @@ export class Command<C extends Context = Context> implements MiddlewareObj<C> {
     match(scope)
-      .with({ type: "default" }, () => this._composer.command(this.names, ...middleware))
+      .with(
+        { type: "default" },
+        () =>
+          this._composer
+            .filter(Command.hasCommand(this.names, optionsObject))
+            .use(...middlewareArray),
+      )
       .with(
         { type: "all_chat_administrators" },
-        () => this._composer.filter(isAdmin).command(this.names, ...middleware),
+        () =>
+          this._composer
+            .filter(Command.hasCommand(this.names, optionsObject))
+            .filter(isAdmin)
+            .use(...middlewareArray),
+      )
+      .with(
+        { type: "all_private_chats" },
+        () =>
+          this._composer
+            .filter(Command.hasCommand(this.names, optionsObject))
+            .chatType("private")
+            .use(
+              ...middlewareArray,
+            ),
       )
-      .with({ type: "all_private_chats" }, () => this._composer.chatType("private").command(this.names, ...middleware))
       .with(
         { type: "all_group_chats" },
-        () => this._composer.chatType(["group", "supergroup"]).command(this.names, ...middleware),
+        () =>
+          this._composer
+            .filter(Command.hasCommand(this.names, optionsObject))
+            .chatType(["group", "supergroup"])
+            .use(
+              ...middlewareArray,
+            ),
       )
       .with(
         { type: P.union("chat", "chat_administrators"), chat_id: P.not(P.nullish).select() },
         (chatId) =>
-          this._composer.filter((ctx) => ctx.chat?.id === chatId).filter(isAdmin).command(this.names, ...middleware),
+          this._composer.filter(Command.hasCommand(this.names, optionsObject))
+            .filter((ctx) => ctx.chat?.id === chatId)
+            .filter(isAdmin)
+            .use(...middlewareArray),
       )
       .with(
         { type: "chat_member", chat_id: P.not(P.nullish).select("chatId"), user_id: P.not(P.nullish).select("userId") },
         ({ chatId, userId }) =>
-          this._composer.filter((ctx) => ctx.chat?.id === chatId)
+          this._composer
+            .filter(Command.hasCommand(this.names, optionsObject))
+            .filter((ctx) => ctx.chat?.id === chatId)
             .filter((ctx) => ctx.from?.id === userId)
-            .command(this.names, ...middleware),
+            .use(...middlewareArray),
       );
 
     this._scopes.push(scope);
 
     return this;
   }
 
-  public localize(languageCode: string, name: string, description: string) {
-    this._languages.set(languageCode, { name, description });
+  public static hasCommand(command: MaybeArray<string | RegExp>, options: CommandOptions) {
+    return (ctx: Context) => {
+      if (!ctx.has(":text")) return false;
+      if (options.matchOnlyAtStart && !ctx.msg.text.startsWith(options.prefix)) return false;
+
+      const commandNames = ensureArray(command).map((name) => new RegExp(name));
```

Comment:
It is not possible to have commands that are exact string matches?

Reply:
It is now. Also, now there's a test that guarantees this works.